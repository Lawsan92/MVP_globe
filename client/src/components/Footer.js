import React from 'react';

const Footer = () => {
  return (
    <section id='footer'>
      <p>Built and Designed By Lawrence Sanzogni</p>
    </section>
  );
};

export default Footer;