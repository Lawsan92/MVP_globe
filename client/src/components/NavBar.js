import React from 'react';
import { Link } from 'react-router-dom';

const NavBar = () =>  {

  return (
    <div id='nav-bar'>
      <div id='logo'><img src='https://res.cloudinary.com/darp0mj9i/image/upload/v1668059703/icons/Screen_Shot_2022-11-01_at_15.45.36_kzwlv1_sjvqta.svg'/></div>
      {/* <div id='search-bar'>
        <input type='text' placeholder='search...'></input>
        <button id='search-button' onClick={() => {alert('clicked!')}}>
          <img id ='mag' src='https://res.cloudinary.com/darp0mj9i/image/upload/v1666116514/samples/magnifying-glass-svgrepo-com_nflk4y.svg'></img>
        </button>
      </div> */}
      <div id='nav-links'>
      <Link to='/' id='link' className='link-login'>
          <p>Home</p>
        </Link>
        <Link to='/login' id='link' className='link-login'>
          <p>Login</p>
        </Link>
      </div>
    </div>
  )
};

export default NavBar;