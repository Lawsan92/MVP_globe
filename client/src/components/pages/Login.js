import React from 'react';
import { Link } from 'react-router-dom';
import { NavBar, Footer } from'../Imports.js';


const Login = ({ userAuth, getUserAuth, userLogin, authorized, userRegister }) => {
  return (
    <div id='login'>
      <NavBar userAuth={userAuth} getUserAuth={getUserAuth} authorized={authorized} />
      <section id='login-main'>
        <form>
          <input type='text' placeholder='username' onChange={() => {getUserAuth({...userAuth, username: event.target.value})}}/>
          <input type='text' placeholder='password'onChange={() => {getUserAuth({...userAuth, password: event.target.value})}}/>
          <input type='submit' value='Login' onClick={() => {event.preventDefault(); userLogin()}}/>
          <input type='submit' value='Register' onClick={() => {event.preventDefault(); userRegister();}}/>
        </form>
      </section>
      <Footer/>
    </div>
  )
};

export default Login;
