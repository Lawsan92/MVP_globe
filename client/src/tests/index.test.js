import renderer from 'react-test-renderer';
import Login from '../components/pages/Login.js';
import Footer from '../components/Footer.js';
import NavBar from '../components/NavBar.js';

it('renders <Footer> correctly', () => {
  const tree = renderer
    .create(<Footer/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders <NavBar> correctly', () => {
  const tree = renderer
    .create(<NavBar/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});


// it('renders <Login> correctly', () => {
//   const tree = renderer
//     .create(<Login/>)
//     .toJSON();
//   expect(tree).toMatchSnapshot();
// });

