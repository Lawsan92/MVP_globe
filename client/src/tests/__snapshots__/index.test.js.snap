// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders <Footer> correctly 1`] = `
<section
  id="footer"
>
  <p>
    Built and Designed By Lawrence Sanzogni
  </p>
</section>
`;
